lib/WebService/Client.pm

```diff
@@ -103,6 +100,26 @@ sub log {
     $self->logger->$log_method($msg);
 }
 
+sub prepare_response {
+    my ($self, $res, $req, $num_attempts) = @_;
+    Moo::Role->apply_roles_to_object($res, 'HTTP::Response::Stringable');
+    return;
+}
+
+sub parse_response {
```

Comment:
I'm not sure about this method. Can you give me an example of how a consumer of the WS::Client would use this method? I think the `req` method being exposed may already may give you want you want.